pkg/network/ebpf/c/protocols/classification/protocol-classification.h

```diff
@@ -161,28 +161,43 @@ __maybe_unused static __always_inline void protocol_classifier_entrypoint(struct
     protocol_t app_layer_proto = get_protocol_from_stack(protocol_stack, LAYER_APPLICATION);
 
     tls_record_header_t tls_hdr = {0};
+    classification_prog_t tail_call = CLASSIFICATION_TLS_CLIENT_PROG;
 
     if ((app_layer_proto == PROTOCOL_UNKNOWN || app_layer_proto == PROTOCOL_POSTGRES) && is_tls(skb, skb_info.data_off, skb_info.data_end, &tls_hdr)) {
         protocol_stack = get_or_create_protocol_stack(&classification_ctx->tuple);
         if (!protocol_stack) {
             return;
         }
         // TLS classification
+        update_protocol_information(classification_ctx, protocol_stack, PROTOCOL_TLS);
         if (tls_hdr.content_type != TLS_HANDSHAKE) {
-            // We can't classify TLS encrypted traffic further, so return early
-            update_protocol_information(classification_ctx, protocol_stack, PROTOCOL_TLS);
+            // If the TLS record is not a handshake, we can stop here as we've already marked the protocol as TLS
+            // and there is no need to look for additional handshake tags
             return;
         }
 
         // Parse TLS handshake payload
         tls_info_t *tags = get_or_create_tls_enhanced_tags(&classification_ctx->tuple);
         if (tags) {
             // The packet is a TLS handshake, so trigger tail calls to extract metadata from the payload
-            goto next_program;
+            __u32 offset = classification_ctx->skb_info.data_off + sizeof(tls_record_header_t);
+            __u32 data_end = classification_ctx->skb_info.data_end;
+            if (is_tls_handshake_client_hello(skb, offset, data_end)) {
+                goto tls_handshake_inspect;
```

Comment:
Any reason to use `goto`, rather than calling `bpf_tail_call_compat` right here?

Reply:
I tried that on a previous commit, but it had the same error output as far as I can tell